lib/OnyxUtils.ts

```diff
@@ -439,18 +440,18 @@ function isSafeEvictionKey(testKey: OnyxKey): boolean {
 }
 
 /**
- * It extracts the non-numeric collection identifier of a given key.
+ * Extracts the collection identifier of a given collection member key.
  *
  * For example:
  * - `getCollectionKey("report_123")` would return "report_"
  * - `getCollectionKey("report_")` would return "report_"
  * - `getCollectionKey("report_-1_something")` would return "report_"
  * - `getCollectionKey("sharedNVP_user_-1_something")` would return "sharedNVP_user_"
  *
- * @param {OnyxKey} key - The key to process.
- * @return {string} The plain collection key.
+ * @param key - The collection key to process.
+ * @returns The plain collection key or throws an Error if the key is not a collection one.
  */
-function getCollectionKey(key: OnyxKey): string {
+function getCollectionKey(key: CollectionKey): string {
```

Comment:
is this correct to use `CollectionKey` if it might not be a collection key and we will throw in that case?

Reply:
We try to avoid the situation by putting a stricter type in the function, but someone can still bypass this and pass an incorrect key, which will lead to an Error being thrown.

Reply:
We have similar strict typing for `splitCollectionMemberKey`